sdk/src/main/scala/com/horizen/account/state/AccountStateViewGasTracked.scala

```diff
@@ -0,0 +1,127 @@
+package com.horizen.account.state
+
+import com.horizen.evm.ResourceHandle
+import com.horizen.evm.interop.EvmLog
+
+import java.math.BigInteger
+
+/**
+ * Wrapper for AccountStateView to help with tracking gas consumption.
+ * @param view
+ *   instance of BaseAccountStateView
+ * @param gas
+ *   GasPool instance to deduct gas from
+ */
+class AccountStateViewGasTracked(view: BaseAccountStateView, gas: GasPool) extends BaseAccountStateView {
+  override def getStateDbHandle: ResourceHandle = view.getStateDbHandle
+
+  override def addAccount(address: Array[Byte], codeHash: Array[Byte]): Unit = view.addAccount(address, codeHash)
+
+  override def accountExists(address: Array[Byte]): Boolean = {
+    gas.subGas(GasUtil.ExtcodeHashGasEIP1884)
+    view.accountExists(address)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def isEoaAccount(address: Array[Byte]): Boolean = {
+    gas.subGas(GasUtil.ExtcodeHashGasEIP1884)
+    view.isEoaAccount(address)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def isSmartContractAccount(address: Array[Byte]): Boolean = {
+    gas.subGas(GasUtil.ExtcodeHashGasEIP1884)
+    view.isSmartContractAccount(address)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def getCodeHash(address: Array[Byte]): Array[Byte] = {
+    gas.subGas(GasUtil.ExtcodeHashGasEIP1884)
+    view.getCodeHash(address)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def getCode(address: Array[Byte]): Array[Byte] = {
+    gas.subGas(GasUtil.ExtcodeHashGasEIP1884)
+    view.getCode(address)
+  }
+
+  override def getNonce(address: Array[Byte]): BigInteger = view.getNonce(address)
+
+  override def increaseNonce(address: Array[Byte]): Unit = view.increaseNonce(address)
+
+  @throws(classOf[OutOfGasException])
+  override def getBalance(address: Array[Byte]): BigInteger = {
+    gas.subGas(GasUtil.BalanceGasEIP1884)
+    view.getBalance(address)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def addBalance(address: Array[Byte], amount: BigInteger): Unit = {
+    gas.subGas(GasUtil.BalanceGasEIP1884)
+    view.addBalance(address, amount)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def subBalance(address: Array[Byte], amount: BigInteger): Unit = {
+    gas.subGas(GasUtil.BalanceGasEIP1884)
+    view.subBalance(address, amount)
+  }
+
+  @throws(classOf[OutOfGasException])
+  override def getAccountStorage(address: Array[Byte], key: Array[Byte]): Array[Byte] = {
+    gas.subGas(GasUtil.GasTBD)
```

Comment:
All `gas.subGas(GasUtil.GasTBD)` must be replaces with the proper values according to the eth one +-.